homeassistant/components/frigidaire/climate.py

```diff
@@ -0,0 +1,253 @@
+        client = frigidaire.Frigidaire(username, password)
+        appliances = client.get_appliances()
+        return [FrigidaireClimate(client, appliance) for appliance in appliances]
+
+    entities = await hass.async_add_executor_job(
+        get_entities, entry.data["username"], entry.data["password"]
+    )
+
+    async_add_entities(entities)
+
+
+class FrigidaireClimate(ClimateEntity):
+    """Representation of a Frigidaire appliance."""
+
+    def __init__(self, client, appliance):
+        """Build FrigidaireClimate.
+
+        data: initially-fetched data.
+        """
+
+        self._client = client
+        self._appliance = appliance
+        self._details = None
+        self._available = False
+        self.update()
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        convert = {
+            frigidaire.Unit.FAHRENHEIT.value: TEMP_FAHRENHEIT,
+            frigidaire.Unit.CELSIUS.value: TEMP_CELSIUS,
+        }
+
+        unit = self._details.for_code(
+            frigidaire.HaclCode.TEMPERATURE_REPRESENTATION
+        ).string_value
+
+        return convert[unit]
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return (
+            self._details.for_code(frigidaire.HaclCode.TARGET_TEMPERATURE)
+            .containers.for_id(frigidaire.ContainerId.TEMPERATURE)
+            .number_value
+        )
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    @property
+    def hvac_mode(self):
+        """Return current operation ie. heat, cool, idle."""
+        convert = {
+            frigidaire.Mode.OFF.value: HVAC_MODE_OFF,
+            frigidaire.Mode.COOL.value: HVAC_MODE_COOL,
+            frigidaire.Mode.FAN.value: HVAC_MODE_FAN_ONLY,
+            frigidaire.Mode.ECO.value: HVAC_MODE_AUTO,
+        }
+
+        frigidaire_mode = self._details.for_code(
+            frigidaire.HaclCode.AC_MODE
+        ).number_value
+
+        return convert[frigidaire_mode]
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return (
+            self._details.for_code(frigidaire.HaclCode.AMBIENT_TEMPERATURE)
+            .containers.for_id(frigidaire.ContainerId.TEMPERATURE)
+            .number_value
+        )
+
+    @property
+    def hvac_modes(self):
+        """List of available operation modes."""
+        return [
+            HVAC_MODE_OFF,
+            HVAC_MODE_COOL,
+            HVAC_MODE_AUTO,
+            HVAC_MODE_FAN_ONLY,
+        ]
```

Comment:
This can be set as an entity class attribute, set in the __init__, making the code more compact:
<https://developers.home-assistant.io/docs/core/entity#entity-class-attributes>